member-service/src/main/java/com/hedvig/memberservice/aggregates/MemberAggregate.kt

```diff
@@ -305,43 +305,45 @@ class MemberAggregate() {
     }
 
     private fun applyMemberIdentifiedEventFromZignSecSignCommand(cmd: ZignSecSignCommand) {
+        val request = objectMapper.readValue(cmd.provideJsonResponse, ZignSecNotificationRequest::class.java)
+        val idProvider = request.identity?.idProviderName ?: return
+        if (member.firstName == cmd.firstName &&
+            member.lastName == cmd.lastName &&
+            member.ssn == cmd.personalNumber) return
         apply(
             MemberIdentifiedEvent(
                 cmd.id,
                 MemberIdentifiedEvent.NationalIdentification(
                     cmd.personalNumber,
-                    when (cmd.zignSecAuthMarket) {
-                        ZignSecAuthenticationMarket.NORWAY -> MemberIdentifiedEvent.Nationality.NORWAY
-                        ZignSecAuthenticationMarket.DENMARK -> MemberIdentifiedEvent.Nationality.DENMARK
-                    }
+                    MemberIdentifiedEvent.Nationality.fromCountryCode(cmd.zignSecAuthMarket.countryCode)
                 ),
-                cmd.zignSecAuthMarket.toMemberIdentifiedEventIdentificationMethod(),
+                "com.zignsec:$idProvider",
                 cmd.firstName,
                 cmd.lastName
             )
         )
     }
 
     @CommandHandler
-    fun handle(command: ZignSecSuccessfulAuthenticationCommand) {
+    fun handle(command: SuccessfulAuthenticationCommand) {
+        if (member.firstName == command.identity.firstName &&
+            member.lastName == command.identity.lastName &&
+            member.ssn == command.identity.nationalIdentifier) return
```

Comment:
Depending on how we treat the "MemberIdentifiedEvent" concept, maybe we can have another approach here. The code is a bit "axon-y" I could maybe elaborate a bit more over CoScreen what I mean if it's cryptic 😄 

```suggestion
        if (isIdentified) return
```